arangod/RestServer/DatabaseFeature.h

```diff
@@ -139,7 +139,12 @@ class DatabaseFeature : public ArangodFeature {
   VocbasePtr useDatabase(std::string_view name) const;
   VocbasePtr useDatabase(TRI_voc_tick_t id) const;
 
-  TRI_vocbase_t* lookupDatabase(std::string_view name) const;
+  // look up a database by name. note: the caller must make sure that the
+  // returned vocbase pointer remains valid (i.e. vocbase is not deleted
+  // concurrently while the returned pointer is used).
+  // this is a potentially unsafe API. if in doubt, prefer using
+  // `useDatabase(...)`, which is safe.
+  [[deprecated]] TRI_vocbase_t* lookupDatabase(std::string_view name) const;
```

Comment:
Why do not just remove?

Reply:
There are still some callers left, which seem to need the raw pointer. But these seem to be ok.
The `lookupDatabase` method should just not be used for any _new_ code.